### PR TITLE
#8295 Initial Yandex.Cloud monitoring

### PR DESCRIPTION
**Pull request by peter-volkov**
### Required for all PRs:

- [x] Signed [CLA](https://influxdata.com/community/cla/).
- [x] Associated README.md updated.
- [x] Has appropriate unit tests.

### CODE REVIEW

File: plugins/outputs/yandex_cloud_monitoring/yandex_cloud_monitoring.go
Comment (p-zak) [inline plugins/outputs/yandex_cloud_monitoring/yandex_cloud_monitoring.go:21]: Ideally all the fields settable from toml should have the toml tags defined.

File: plugins/outputs/yandex_cloud_monitoring/yandex_cloud_monitoring.go
Comment (p-zak) [inline plugins/outputs/yandex_cloud_monitoring/yandex_cloud_monitoring.go:25]: `MetadataTokenUrl`, `MetadataFolderUrl` and `Service` are undocumented in sample config. Is this desired behavior?

File: plugins/outputs/yandex_cloud_monitoring/yandex_cloud_monitoring.go
Comment (p-zak) [inline plugins/outputs/yandex_cloud_monitoring/yandex_cloud_monitoring.go:93]: `if a.Timeout.Duration <= 0 {` would be better.

File: plugins/outputs/yandex_cloud_monitoring/yandex_cloud_monitoring.go
Comment (peter-volkov) [reply plugins/outputs/yandex_cloud_monitoring/yandex_cloud_monitoring.go:25]: Yes. Non-default service can be set only in internal use cases and we plan to use it,
though for public clients it should always be set to default value "custom".
This is stated in docs.

File: plugins/outputs/yandex_cloud_monitoring/yandex_cloud_monitoring.go
Comment (peter-volkov) [reply plugins/outputs/yandex_cloud_monitoring/yandex_cloud_monitoring.go]: Done

File: plugins/outputs/yandex_cloud_monitoring/yandex_cloud_monitoring.go
Comment (peter-volkov) [reply plugins/outputs/yandex_cloud_monitoring/yandex_cloud_monitoring.go]: Sure

**Comment by peter-volkov** [comment]
@p-zak can you please take another look?

**Comment by zak-pawel** [comment]
> @p-zak can you please take another look?

LGTM

**Comment by p-zak** [comment]
> @p-zak can you please take another look?

LGTM!

**Comment by peter-volkov** [comment]
@p-zak thank you. Who can merge the PR?

**Comment by p-zak** [comment]
> @p-zak thank you. Who can merge the PR?

Need to wait for @ssoroka or @reimda

File: plugins/outputs/yandex_cloud_monitoring/README.md
Comment (ssoroka) [inline plugins/outputs/yandex_cloud_monitoring/README.md:21]: missing some supported fields here.
commented values should be set to whatever the default values are.

File: plugins/outputs/yandex_cloud_monitoring/yandex_cloud_monitoring.go
Comment (ssoroka) [inline plugins/outputs/yandex_cloud_monitoring/yandex_cloud_monitoring.go:124]: If you add `Log telegraf.Logger `toml:"-"` to the struct, you can use the Telegraf logger that knows about plugin names and log levels. In tests, set it to `telegraf.Logger{}`

File: plugins/outputs/yandex_cloud_monitoring/yandex_cloud_monitoring.go
Comment (ssoroka) [inline plugins/outputs/yandex_cloud_monitoring/yandex_cloud_monitoring.go:172]: body is nil here. there's no need to post it separately, and I don't see how this solves the problem where the jsonBytes is larger than maxRequestBodySize.

File: plugins/outputs/yandex_cloud_monitoring/yandex_cloud_monitoring.go
Comment (ssoroka) [inline plugins/outputs/yandex_cloud_monitoring/yandex_cloud_monitoring.go:29]: Nothing special is done with the IAMToken, it's just set if it's blank. you can merge these two.

File: plugins/outputs/yandex_cloud_monitoring/yandex_cloud_monitoring_test.go
Comment (ssoroka) [inline plugins/outputs/yandex_cloud_monitoring/yandex_cloud_monitoring_test.go:22]: use `require.NoErr(t, err)`

**Review by ssoroka — CHANGES_REQUESTED** [review]

**Comment by ssoroka** [comment]
Looking pretty good. I have a few more change requests and after that I'd be happy to merge.

File: plugins/outputs/yandex_cloud_monitoring/yandex_cloud_monitoring.go
Comment (peter-volkov) [reply plugins/outputs/yandex_cloud_monitoring/yandex_cloud_monitoring.go]: Done

File: plugins/outputs/yandex_cloud_monitoring/README.md
Comment (peter-volkov) [reply plugins/outputs/yandex_cloud_monitoring/README.md:18]: Done

File: plugins/outputs/yandex_cloud_monitoring/yandex_cloud_monitoring.go
Comment (peter-volkov) [reply plugins/outputs/yandex_cloud_monitoring/yandex_cloud_monitoring.go]: Thanks. Done

File: plugins/outputs/yandex_cloud_monitoring/yandex_cloud_monitoring_test.go
Comment (peter-volkov) [reply plugins/outputs/yandex_cloud_monitoring/yandex_cloud_monitoring_test.go]: Done

File: plugins/outputs/yandex_cloud_monitoring/yandex_cloud_monitoring.go
Comment (peter-volkov) [reply plugins/outputs/yandex_cloud_monitoring/yandex_cloud_monitoring.go]: We do not implement special limit for POST at server side now. So I removed this part for now.

File: plugins/outputs/yandex_cloud_monitoring/README.md
Comment (ssoroka) [inline plugins/outputs/yandex_cloud_monitoring/README.md:11]: I'm not sure the lower timeout is beneficial here. With some lag you could see a lot of metric retransmission.

File: plugins/outputs/yandex_cloud_monitoring/yandex_cloud_monitoring.go
Comment (ssoroka) [inline plugins/outputs/yandex_cloud_monitoring/yandex_cloud_monitoring.go:118]: these should probably be Debugf, as under normal operation the plugins should be rather quiet.

File: plugins/outputs/yandex_cloud_monitoring/yandex_cloud_monitoring.go
Comment (ssoroka) [inline plugins/outputs/yandex_cloud_monitoring/yandex_cloud_monitoring.go:27]: some of the linters will complain about this.
```suggestion
	MetadataTokenURL       string
	MetadataFolderURL      string
```

**Review by ssoroka — COMMENTED** [review]
Looks good. I've requested a few small edits, and then we can get this merged.

File: plugins/outputs/yandex_cloud_monitoring/yandex_cloud_monitoring.go
Comment (peter-volkov) [reply plugins/outputs/yandex_cloud_monitoring/yandex_cloud_monitoring.go:118]: I have intentionally made this info-level message -- this is being printed only once during plugin initialization. Seems not so verbose to me. Do you insist?

File: plugins/outputs/yandex_cloud_monitoring/README.md
Comment (peter-volkov) [reply plugins/outputs/yandex_cloud_monitoring/README.md:11]: 5s seems pretty huge timeout for monitoring http write handler, isn't it?
It was 5s before. I just changed this config comments to make it have default values as you asked.

File: plugins/outputs/yandex_cloud_monitoring/yandex_cloud_monitoring.go
Comment (ssoroka) [reply plugins/outputs/yandex_cloud_monitoring/yandex_cloud_monitoring.go:118]: Ok, feel free to leave it.

File: plugins/outputs/yandex_cloud_monitoring/yandex_cloud_monitoring.go
Comment (peter-volkov) [reply plugins/outputs/yandex_cloud_monitoring/yandex_cloud_monitoring.go:27]: Do you mean lexical order of the properties? 
Ok I will change this

File: plugins/outputs/yandex_cloud_monitoring/README.md
Comment (ssoroka) [reply plugins/outputs/yandex_cloud_monitoring/README.md:11]: I believe this is used as the write deadline internally, so it doesn't just refer to 5s idle waiting for the response, but the entire process of writing the metrics in the request. If metrics ever take more than 5s to write, this will retransmit them in a loop until they're dropped, even if the receiver read them.

File: plugins/outputs/yandex_cloud_monitoring/yandex_cloud_monitoring.go
Comment (ssoroka) [reply plugins/outputs/yandex_cloud_monitoring/yandex_cloud_monitoring.go:27]: no, just that URL is typically capitalized

File: plugins/outputs/yandex_cloud_monitoring/README.md
Comment (peter-volkov) [reply plugins/outputs/yandex_cloud_monitoring/README.md:11]: Alright. I will increase the timeout

File: plugins/outputs/yandex_cloud_monitoring/yandex_cloud_monitoring.go
Comment (peter-volkov) [reply plugins/outputs/yandex_cloud_monitoring/yandex_cloud_monitoring.go:27]: Oh. That's right, thank you

Comment (peter-volkov) [reply plugins/outputs/yandex_cloud_monitoring/yandex_cloud_monitoring.go]: Done

File: plugins/outputs/yandex_cloud_monitoring/README.md
Comment (peter-volkov) [reply plugins/outputs/yandex_cloud_monitoring/README.md]: Done

**Review by ssoroka — APPROVED** [review]

**Comment by ssoroka** [comment]
Thanks! Will merge as soon as the tests pass

**Pull request merged**
